src/rulesets/Base/base.jl

```diff
@@ -146,14 +146,21 @@ end
 @scalar_rule x \ y (-(Ω / x), one(y) / x)
 
 function frule((_, ẏ), ::typeof(identity), x)
-    return (x, ẏ)
+    return x, ẏ
 end
 
 function rrule(::typeof(identity), x)
-    function identity_pullback(ȳ)
-        return (NoTangent(), ȳ)
-    end
-    return (x, identity_pullback)
+    identity_pullback(ȳ)= NoTangent(), ȳ
+    return x, identity_pullback
+end
+
+function frule((_, ẏ), ::typeof(copy), x)
```

Comment:
why do we also want the frule?

Reply:
I don't know, I was just mimicking identity, should I remove it?

Reply:
🤷 maybe, no code is the best code :)